python/lsst/ts/m2com/mock/mock_model.py

```diff
@@ -383,6 +394,58 @@ def is_actuator_force_out_limit(
 
         return is_out_limit, error_code, limit_switches_retract, limit_switches_extend
 
+    def is_force_error_tangent_out_limit(self) -> tuple[bool, MockErrorCode]:
+        """The force error of tangent link is out of limit or not.
+
+        This function is translated from TangentLoadCellFaultDetection.vi in
+        ts_mtm2 LabVIEW project.
+
+        Returns
+        -------
+        is_out_limit : `bool`
+            True if the force error is out of limit. Otherwise, False.
+        error_code : enum `MockErrorCode`
+            Error code.
+        """
+
+        # Check four fault conditions
+        force_error_tangent = self._force_error_tangent
+
+        is_out_limit_total_weight = (
+            np.abs(force_error_tangent["weight"]) >= TANGENT_LINK_TOTAL_WEIGHT_ERROR
+        )
+        is_out_limit_theta_z = (
+            np.abs(force_error_tangent["sum"]) >= TANGENT_LINK_THETA_Z_MOMENT
+        )
+
+        force_error_individual = np.array(force_error_tangent["force"])
+
+        # Tangent link: A1, A4
+        is_out_limit_non_load = np.any(
+            np.abs(force_error_individual[[0, 3]]) >= TANGENT_LINK_NON_LOAD_BEARING_LINK
+        )
+
+        # Tangent link: A2, A3, A5, A6
+        is_out_limit_load = np.any(
+            np.abs(force_error_individual[[1, 2, 4, 5]])
+            >= TANGENT_LINK_LOAD_BEARING_LINK
+        )
+
+        # Decide the error code
+        is_out_limit = (
+            is_out_limit_total_weight
+            or is_out_limit_theta_z
+            or is_out_limit_non_load
+            or is_out_limit_load
+        )
+        error_code = (
+            MockErrorCode.TangentLoadCellFault
+            if is_out_limit
+            else MockErrorCode.NoError
+        )
+
+        return is_out_limit, error_code
```

Comment:
Do you really need to return the boolean part? It seems like returning the error code alone would be sufficient.

Reply:
This is to help the `if` loop in `MockServer._check_error_force()`. Thanks!

```
        # Check the force error of tangent links
        (
            is_out_limit_force_error_tangent,
            error_code_force_error_tangent,
        ) = self.model.is_force_error_tangent_out_limit()

        if is_out_limit_force_error_tangent:
            self.model.fault(error_code_force_error_tangent)
```